hsclient/hydroshare.py

```diff
@@ -314,15 +393,167 @@ def refresh(self) -> None:
         self._parsed_files = None
         self._parsed_aggregations = None
         self._parsed_checksums = None
+        self._main_file_path = None
+
+    def delete(self) -> None:
+        """Deletes this aggregation from HydroShare"""
+        path = urljoin(
+            self._hsapi_path,
+            "functions",
+            "delete-file-type",
+            self.metadata.type.value + "LogicalFile",
+            self.main_file_path,
+        )
+        self._hs_session.delete(path, status_code=200)
+        self.refresh()
 
-    def as_series(self, series_id: str, agg_path: str = None) -> Dict[int, pandas.Series]:
-        """
-        Creates a pandas Series object out of an aggregation of type TimeSeries.
-        :param series_id: The series_id of the timeseries result to be converted to a Series object.
-        :param agg_path: Not required.  Include this parameter to avoid downloading the aggregation if you already have
-        it downloaded locally.
-        :return: A pandas.Series object
-        """
+
+class DataObjectSupportingAggregation(Aggregation):
```

Comment:
I like the approach of different class implementations for each aggregation type.